### PR TITLE
fix podutils images built by bazel

### PR DESCRIPTION
**Pull request by BenTheElder**
switch to `container_image`, which is less magical, and just place the binaries in root.
fixes: https://github.com/kubernetes/test-infra/issues/7947

### CODE REVIEW

**Comment by BenTheElder** [comment]
```console
$ bazel build //prow/cmd/initupload:image.tar
INFO: Analysed target //prow/cmd/initupload:image.tar (1 packages loaded).
INFO: Found 1 target...
Target //prow/cmd/initupload:image.tar up-to-date:
  bazel-bin/prow/cmd/initupload/image.tar
INFO: Elapsed time: 1.879s, Critical Path: 1.62s
INFO: 6 processes: 6 linux-sandbox.
INFO: Build completed successfully, 8 total actions

$ docker load -i bazel-bin/prow/cmd/initupload/image.tar
1b0f8600cab8: Loading layer [==================================================>]  15.56MB/15.56MB
The image bazel/prow/cmd/initupload:image already exists, renaming the old one with ID sha256:f74c1892d6fb37db3b5d71e7031d34b8bf64a12e7d7ff6e3f46da1758ec32edb to empty string
Loaded image: bazel/prow/cmd/initupload:image

$ docker run --rm -it --entrypoint=/bin/sh bazel/prow/cmd/initupload:image
/ # ls
bin         dev         etc         home        initupload  lib         media       mnt         proc        root        run         sbin        srv         sys         tmp         usr         var
/ # /initupload
{"component":"initupload","error":"could not resolve job spec: $JOB_SPEC unset","level":"fatal","msg":"Failed to initialize job","time":"2018-08-09T00:43:18Z"}
```

File: prow/cmd/initupload/BUILD.bazel [hunk 1]
```diff
@@ -1,11 +1,10 @@
-load("@io_bazel_rules_docker//go:image.bzl", "go_image")
+load("@io_bazel_rules_docker//container:image.bzl", "container_image")
 load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
 
-go_image(
+container_image(
     name = "image",
     base = "@alpine-base//image",
-    binary = ":initupload",
-    directory = "/",
```
Comment (ixdy) [inline prow/cmd/initupload/BUILD.bazel]: why did you remove `directory` here but not in the other images?

Comment (BenTheElder) [reply prow/cmd/initupload/BUILD.bazel]: I'm having some trouble with buildifier-on-save eating my changes :confused:, I put this back... fixing

Comment (BenTheElder) [reply prow/cmd/initupload/BUILD.bazel]: fixed, we definitely want that.

**Comment by BenTheElder** [comment]
Agh, racing with buildify-on-save in my editor, should be fixed now.

**Comment by fejta** [comment]
Add this functionality to go_image instead?

**Comment by fejta** [comment]
https://github.com/bazelbuild/rules_docker/blob/26e5ff9a6dc292452de518e2393afe6b80f2d390/lang/image.bzl#L261

https://github.com/bazelbuild/rules_docker/blob/26e5ff9a6dc292452de518e2393afe6b80f2d390/go/image.bzl#L91

**Comment by BenTheElder** [comment]
@fejta we're not leveraging anything special about `go_image`, we're using our own base image, we don't use runfiles, we hardcode that the binaries must be in the root, and we always build with `pure="on"`... it makes sense to just take the base image and deposit the binary in `/`. For that, `container_image` is the right rule.

**Comment by fejta** [comment]
Expedient perhaps but the least useful solution to the next person working with go and bazel

**Comment by BenTheElder** [comment]
I mean the `go_image` rule is WAI, expecting the binary to be in `/foo` as opposed to say `$PATH` is abnormal, but how the podutils work. With `go_image` you are meant to let the rule control the entrypoint, placing the binary, placing the binary's runfiles, etc. We don't use any of that for the podutils.

**Comment by BenTheElder** [comment]
None of the `${lang}_image` rules support controlling the paths in any way.
